### DISCUSSION TITLE
Preload fails for path specified in call function

### DISCUSSION THREAD

**Discussion by GAldegheri** [jspsych/jsPsych#1793] [Q&A] [answered]
Hi, in my experiment I'm trying to preload an image which is set from a call function.
This fails, giving me "Error loading file".
It seems however that the image path is visible from the preloading node, i.e. if I use a on_start function to print it, it's printed correctly.
But it cannot load it for some reason.
Here is a minimal code example that reproduces this issue:
```js
  var stimulus = '';

  var setstimulus = {
    type: 'call-function',
    func: function(){stimulus = 'image.png';}
  };

  var preload_trial = {
    type: 'preload',
    images: [stimulus],
    show_progress_bar: false,
    show_detailed_errors: true,
    on_start: function(){console.log(stimulus);}
  };

  jsPsych.init({
      timeline: [setstimulus, preload_trial]
  });
```
Is this issue known with call functions?

**Comment by jodeleeuw** [comment, marked as answer]
Hi @GAldegheri 

At the time that `preload_trial` is declared `stimulus` is still `''`. The `call-function` trial doesn't run until after the experiment is initialized, at which point the value of `stimulus` is updated. 

To use that updated value for `preload_trial`, you need to use [dynamic parameters](https://www.jspsych.org/overview/dynamic-parameters/).

**Comment by GAldegheri** [comment]
Thanks for your reply, it works!
